src/AppBundle/Resources/javascript/Component/HideAbleComponent.ts

```diff
@@ -0,0 +1,17 @@
+import 'jquery';
+import { Component } from './Component';
+
+export class HideAbleComponent implements Component {
```

Comment:
Hidable is one word, Or is Able actually in hiding here ;) Same goes for your other Able classes. /nitpick